### PR TITLE
Fix download result for empty JSOC searches

### PR DESCRIPTION
**Pull request by dstansby**
See changelog entry and test for context.

Fixes https://github.com/sunpy/sunpy/issues/6395

### CODE REVIEW

File: sunpy/net/jsoc/tests/test_jsoc.py [hunk 1]
```diff
@@ -355,3 +355,15 @@ def test_check_request_keywords(client):
         a.Time('2020/1/1T1:00:36', '2020/1/1T01:00:38'),
         a.jsoc.Series('aia.lev1_euv_12s'), a.jsoc.Keyword("QUALITY") < 2, a.jsoc.Keyword("EXPTIME") > 2)
     assert len(responses) == 7
+
+
+@pytest.mark.remote_data
+def test_empty_response_fetch(client):
+    # Check that results from downloading an empty response are empty
+    response = client.search(
+        a.Time('2022-12-12T00:00:00', '2022-12-12T0:01:00'),
```
Comment (dstansby) [inline sunpy/net/jsoc/tests/test_jsoc.py:364]:
```suggestion
        a.Time('1990-12-12T00:00:00', '1990-12-12T0:01:00'),
```

**Review by nabobalis — APPROVED** [review]

File: sunpy/net/jsoc/tests/test_jsoc.py [hunk 2]
```diff
@@ -355,3 +355,15 @@ def test_check_request_keywords(client):
...
+
+
+@pytest.mark.remote_data
+def test_empty_response_fetch(client):
+    # Check that results from downloading an empty response are empty
+    response = client.search(
+        a.Time('1990-12-12T00:00:00', '1990-12-12T0:01:00'),
+        a.jsoc.Series.hmi_b_720s
+    )
+    assert len(response) == 0
```
Comment (Cadair) [inline sunpy/net/jsoc/tests/test_jsoc.py:367]: Can we not mock a response here to test this behaviour in an offline test?

Something like this should work:
```suggestion
    response = JSOCResponse()
```

Comment (dstansby) [reply sunpy/net/jsoc/tests/test_jsoc.py:367]: We could, but I think it's safer here to test the code in the way a user would interact with it.

Comment (Cadair) [reply sunpy/net/jsoc/tests/test_jsoc.py:367]: I don't think anything that actually involves interacting with JSOC servers is a good test :stuck_out_tongue: 

If we can avoid having to make a call to JSOC then the test is going to be a lot more reliable and less likely to be ignored as a "transient failure".

If you want to have both then ok, but I would rather have one which is offline as well.

Comment (dstansby) [reply sunpy/net/jsoc/tests/test_jsoc.py]: 👍  - I've done as suggested, but left the search in a comment so it's easy to enable an end-to-end test in the future (e.g. if the `vcrpy` stuff ever gets sorted).

**Pull request merged**